### PR TITLE
Clean up code from previous ref-readonly PRs

### LINKED ISSUE #21981: Remove specialcasing for Span and ReadOnlySpan before shipping
We currently assume, regardless of attributes, that Span and ReadOnlySpan are
- ref-like structs
- readonly structs

This is done as a bootstrapping measure to mitigate the situation where some repos have old compilers and some implementations of span types do not have expected annotation.

We should remove the special-casing when before shipping to not set a compat precedent where annotations are unnecessary on span types.

### PR DESCRIPTION
**Pull request by OmarTawfik**
Fixes #21981 - Remove specialcasing for Span and ReadOnlySpan before shipping
Responds to one comment on #22424 

cc @dotnet/roslyn-compiler @VSadov

### CODE REVIEW

File: src/Compilers/CSharp/Test/Semantic/Semantics/RefExtensionMethodsTests.cs [hunk 1]
```diff
@@ -982,7 +982,7 @@ public static void Main()
             var reference = CreateCompilationWithMscorlibAndSystemCore(code).VerifyDiagnostics(
                 // (4,24): error CS8338: The first parameter of the readonly reference extension method 'PrintValue' must be a value type.
                 //     public static void PrintValue<T>(in this T p) where T : System.IComparable
-                Diagnostic(ErrorCode.ERR_RefReadOnlyExtensionMustBeValueType, "PrintValue").WithArguments("PrintValue").WithLocation(4, 24),
+                Diagnostic(ErrorCode.ERR_InExtensionMustBeValueType, "PrintValue").WithArguments("PrintValue").WithLocation(4, 24),
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Semantic/Semantics/RefExtensionMethodsTests.cs]: Could you add a test with a legacy Span (lacking `ref`)? It should no longer be recognized.

**Comment by jcouv** [comment]
Approved for 15.5

**Comment by TyOverby** [comment]
LGTM

File: src/Compilers/CSharp/Test/Semantic/Semantics/RefEscapingTests.cs [hunk 2]
```diff
@@ -2515,7 +2515,7 @@ public void M(ref Span<int> global)
 }
 public static class Extensions
 {
-    public static void Deconstruct(ref readonly this Span<int> self, out Span<int> x, out Span<int> y) => throw null;
+    public static void Deconstruct(in this Span<int> self, out Span<int> x, out Span<int> y) => throw null;
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Semantic/Semantics/RefEscapingTests.cs]: Thanks. Would you mind unskipping the test too?

**Comment by OmarTawfik** [comment]
@VSadov any objections on the test changes?

**Pull request merged**